plugins/node/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -0,0 +1,477 @@
+          if (isWrapped(moduleExports?.prototype?.on)) {
+            this._unwrap(moduleExports.prototype, "on");
+          }
+          this._wrap(
+            moduleExports.prototype,
+            "on",
+            this._patchOn(moduleVersion)
+          );
+          return moduleExports;
+        },
+        (moduleExports, moduleVersion) => {
+          if (isWrapped(moduleExports?.prototype?.on)) {
+            this._unwrap(moduleExports.prototype, "on");
+          }
+          return moduleExports;
+        },
+        [namespaceInstrumentationLegacy, socketInstrumentationLegacy]
+      ),
+    ];
+  }
+
+  override setConfig(config: SocketIoInstrumentationConfig = {}) {
+    return super.setConfig(normalizeConfig(config));
+  }
+
+  private _patchOn(moduleVersion: string) {
+    const self = this;
+    return (original: Function) => {
+      return function (this: any, ev: any, originalListener: Function) {
+        if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+          return original.apply(this, arguments);
+        }
+        if (self._config.onIgnoreEventList?.includes(ev)) {
+          return original.apply(this, arguments);
+        }
+        const wrappedListener = function (this: any, ...args: any[]) {
+          const eventName = ev;
+          const defaultNamespace = "/";
+          const namespace = this.name || this.adapter?.nsp?.name;
+          const destination =
+            namespace === defaultNamespace
+              ? eventName
+              : `${namespace} ${eventName}`;
+          const span: Span = self.tracer.startSpan(
+            `${destination} ${MessagingOperationValues.RECEIVE}`,
+            {
+              kind: SpanKind.CONSUMER,
+              attributes: {
+                [SemanticAttributes.MESSAGING_SYSTEM]: "socket.io",
+                [SemanticAttributes.MESSAGING_DESTINATION]: namespace,
+                [SemanticAttributes.MESSAGING_OPERATION]:
+                  MessagingOperationValues.RECEIVE,
+                [SocketIoInstrumentationAttributes.SOCKET_IO_EVENT_NAME]:
+                  eventName,
+              },
+            }
+          );
+
+          if (self._config.onHook) {
+            safeExecuteInTheMiddle(
+              () =>
+                self._config?.onHook?.(span, { moduleVersion, payload: args }),
+              (e) => {
+                if (e) self._diag.error(`onHook error`, e);
+              },
+              true
+            );
+          }
+          return context.with(trace.setSpan(context.active(), span), () =>
+            self.endSpan(() => originalListener.apply(this, arguments), span)
+          );
+        };
+        return original.apply(this, [ev, wrappedListener]);
+      };
+    };
+  }
+
+  private endSpan(traced: () => any | Promise<any>, span: Span) {
+    try {
+      const result = traced();
+      if (isPromise(result)) {
+        return Promise.resolve(result)
+          .catch((err) => {
+            if (err) {
```

Comment:
I think we don't need the `if(err)` as we are already in the catch part which means there was an error